# Group Assignment (ANN)

This project sought to analyse the data of over 1000 companies using Artificial Neural Network in order to forecast the valuation of another company given specific variables.

### Importing nessesury Module and library

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns; sns.set()

import tensorflow as tf
from tensorflow import keras

import sklearn
from sklearn.preprocessing import StandardScaler
from sklearn.model_selection import train_test_split, cross_val_score
from sklearn.metrics import r2_score, classification_report, confusion_matrix


2022-05-06 11:22:03.290149: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory
2022-05-06 11:22:03.290176: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.


### Importing dataset 

In [2]:
#import diabetes dataset
df = pd.read_csv("/home/md12.islam/ ML_Module/Untitled Folder/Unicorn_Companies.csv" )
#show size of rows and columns
df.shape
#show first 5 rows in dataset
df.head()

Company  Valuation ($B) Date Joined        Country           City  \
0   Bytedance           140.0  04/07/2017          China        Beijing   
1      SpaceX           100.3  12/01/2012  United States      Hawthorne   
2      Stripe            95.0   1/23/2014  United States  San Francisco   
3      Klarna            45.6  12/12/2011         Sweden      Stockholm   
4  Epic Games            42.0  10/26/2018  United States           Cary   

                  Industry                                  Select Inverstors  \
0  Artificial intelligence  Sequoia Capital China, SIG Asia Investments, S...   
1                    Other  Founders Fund, Draper Fisher Jurvetson, Rothen...   
2                  Fintech        Khosla Ventures, LowercaseCapital, capitalG   
3                  Fintech  Institutional Venture Partners, Sequoia Capita...   
4                    Other              Tencent Holdings, KKR, Smash Ventures   

   Founded Year Total Raised Financial Stage  Investors Count  Deal Terms  \
0          2012       $7.44B             IPO               28           8   
1          2002      $6.874B            None               29          12   
2          2010      $2.901B           Asset               39          12   
3          2005      $3.472B        Acquired               56          13   
4          1991      $4.377B        Acquired               25           5   

  Portfolio Exits  
0               5  
1            None  
2               1  
3               1  
4               2

### Data Exploration

In [3]:
df.info()
df.describe(include='all')

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1037 entries, 0 to 1036
Data columns (total 13 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Company            1037 non-null   object 
 1   Valuation ($B)     1037 non-null   float64
 2   Date Joined        1037 non-null   object 
 3   Country            1037 non-null   object 
 4   City               1037 non-null   object 
 5   Industry           1037 non-null   object 
 6   Select Inverstors  1037 non-null   object 
 7   Founded Year       1037 non-null   int64  
 8   Total Raised       1037 non-null   object 
 9   Financial Stage    1037 non-null   object 
 10  Investors Count    1037 non-null   int64  
 11  Deal Terms         1037 non-null   int64  
 12  Portfolio Exits    1037 non-null   object 
dtypes: float64(1), int64(3), object(9)
memory usage: 105.4+ KB


Company  Valuation ($B) Date Joined        Country           City  \
count     1037     1037.000000        1037           1037           1037   
unique    1035             NaN         623             46            256   
top       Bolt             NaN   7/13/2021  United States  San Francisco   
freq         2             NaN           9            536            145   
mean       NaN        3.290154         NaN            NaN            NaN   
std        NaN        7.310122         NaN            NaN            NaN   
min        NaN        1.000000         NaN            NaN            NaN   
25%        NaN        1.070000         NaN            NaN            NaN   
50%        NaN        1.600000         NaN            NaN            NaN   
75%        NaN        3.100000         NaN            NaN            NaN   
max        NaN      140.000000         NaN            NaN            NaN   

       Industry Select Inverstors  Founded Year Total Raised Financial Stage  \
count      1037              1037   1037.000000         1037            1037   
unique       33              1006           NaN          914              10   
top     Fintech              None           NaN         None            None   
freq        205                17           NaN           24             988   
mean        NaN               NaN   1929.251688          NaN             NaN   
std         NaN               NaN    401.499520          NaN             NaN   
min         NaN               NaN      0.000000          NaN             NaN   
25%         NaN               NaN   2011.000000          NaN             NaN   
50%         NaN               NaN   2014.000000          NaN             NaN   
75%         NaN               NaN   2016.000000          NaN             NaN   
max         NaN               NaN   2021.000000          NaN             NaN   

        Investors Count   Deal Terms Portfolio Exits  
count       1037.000000  1037.000000            1037  
unique              NaN          NaN               5  
top                 NaN          NaN            None  
freq                NaN          NaN             988  
mean          14.418515     3.006750             NaN  
std            9.951832     2.189602             NaN  
min            0.000000     0.000000             NaN  
25%            8.000000     1.000000             NaN  
50%           13.000000     3.000000             NaN  
75%           19.000000     4.000000             NaN  
max           91.000000    19.000000             NaN

* this dataset has total 13 columns and 1037 row.
* only Valuation ($B), Investors Count,Deal Terms  are numerical value rests
* rest are objects(needs to be modified to numeric)

In [4]:
# Check if any column has null variables
df.isnull().sum()

Company              0
Valuation ($B)       0
Date Joined          0
Country              0
City                 0
Industry             0
Select Inverstors    0
Founded Year         0
Total Raised         0
Financial Stage      0
Investors Count      0
Deal Terms           0
Portfolio Exits      0
dtype: int64

In [5]:
#Check if any column has None Value

for column in df.columns:
    if 'None' in df[column].unique():
        print(column)

Select Inverstors
Total Raised
Financial Stage
Portfolio Exits


/tmp/ipykernel_613377/2188549246.py:4: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  if 'None' in df[column].unique():


In [6]:
#Check if any column has None Value

for column in df.columns:
    if 0 in df[column].unique():
        print(column)

Founded Year
Investors Count
Deal Terms


In [7]:
# Exploring Industry columns
print(df["Industry"].value_counts())

Fintech                                                               205
Internet software & services                                          192
E-commerce & direct-to-consumer                                       107
Artificial intelligence                                                71
Health                                                                 69
Supply chain, logistics, & delivery                                    57
Other                                                                  56
Cybersecurity                                                          49
Data management & analytics                                            41
Mobile & telecommunications                                            37
Hardware                                                               33
Auto & transportation                                                  29
Edtech                                                                 28
Consumer & retail                     

column( Industry) has few unique value which accured only once, so we are going to get rid of those rare row so that our Model doesn't gets confused.

In [8]:
uselessnames = ["Finttech","Sequoia Capital China, Shunwei Capital Partners, Qualgro","B Capital Group, Monk's Hill Ventures, Dynamic Parcel Distribution","Andreessen Horowitz, DST Global, IDG Capital","Vertex Ventures SE Asia, Global Founders Capital, Visa Ventures","Mundi Ventures, Doqling Capital Partners, Activant Capital","SingTel Innov8, Alpha JWC Ventures, Golden Gate Ventures","Dragonfly Captial, Qiming Venture Partners, DST Global","Sequoia Capital China, ING, Alibaba Entrepreneurs Fund","Sequoia Capital, Thoma Bravo, Softbank","500 Global, Rakuten Ventures, Golden Gate Ventures","Hopu Investment Management, Boyu Capital, DC Thomson Ventures","Vision Plus Capital, GSR Ventures, ZhenFund","Jungle Ventures, Accel, Venture Highway","Tiger Global Management, Tiger Brokers, DCM Ventures","Kuang-Chi","Temasek, Guggenheim Investments, Qatar Investment Authority"]
for i in uselessnames:
    df.drop( df[ df["Industry"] == i ].index , inplace=True)

In [9]:
print(df["Industry"].value_counts())

Fintech                                205
Internet software & services           192
E-commerce & direct-to-consumer        107
Artificial intelligence                 71
Health                                  69
Supply chain, logistics, & delivery     57
Other                                   56
Cybersecurity                           49
Data management & analytics             41
Mobile & telecommunications             37
Hardware                                33
Auto & transportation                   29
Edtech                                  28
Consumer & retail                       25
Travel                                  14
Artificial Intelligence                  7
Name: Industry, dtype: int64


Same Process for the column(Finalcial Stage)

In [10]:
print(df["Financial Stage"].value_counts())

None           973
Acquired        21
Divestiture      8
IPO              7
Acq              6
Asset            1
Take             1
Management       1
Reverse          1
Corporate        1
Name: Financial Stage, dtype: int64


In [11]:
uselessnames = ["Asset","Take","Management","Reverse","Corporate"]
for i in uselessnames:
    df.drop( df[ df["Financial Stage"] == i ].index , inplace=True)

In [12]:
print(df["Financial Stage"].value_counts())

None           973
Acquired        21
Divestiture      8
IPO              7
Acq              6
Name: Financial Stage, dtype: int64


### Data Modifidation

column(Date Joined, Founded year) has Date object type data.
which is not good for machine learning.
We are going to convert this data into age data as two new column as ("Age" and "Unicorn Age")


In [13]:
from datetime import datetime, date

df['Date Joined']

0       04/07/2017
1       12/01/2012
3       12/12/2011
4       10/26/2018
5       01/08/2018
           ...    
1032     2/22/2022
1033     2/23/2022
1034     2/23/2022
1035     2/23/2022
1036     2/24/2022
Name: Date Joined, Length: 1015, dtype: object

In [14]:
df['Date Joined'] = pd.to_datetime(df['Date Joined'], format='%m/%d/%Y')
# Create 'year_joined column'
df['year joined'] = df['Date Joined'].dt.year

In [15]:
df['age'] = (2022-(df['Founded Year']))
df['Unicorn age'] = (2022-(df['year joined']))
df.head()

Company  Valuation ($B) Date Joined        Country         City  \
0   Bytedance           140.0  2017-04-07          China      Beijing   
1      SpaceX           100.3  2012-12-01  United States    Hawthorne   
3      Klarna            45.6  2011-12-12         Sweden    Stockholm   
4  Epic Games            42.0  2018-10-26  United States         Cary   
5       Canva            40.0  2018-01-08      Australia  Surry Hills   

                       Industry  \
0       Artificial intelligence   
1                         Other   
3                       Fintech   
4                         Other   
5  Internet software & services   

                                   Select Inverstors  Founded Year  \
0  Sequoia Capital China, SIG Asia Investments, S...          2012   
1  Founders Fund, Draper Fisher Jurvetson, Rothen...          2002   
3  Institutional Venture Partners, Sequoia Capita...          2005   
4              Tencent Holdings, KKR, Smash Ventures          1991   
5  Sequoia Capital China, Blackbird Ventures, Mat...          2012   

  Total Raised Financial Stage  Investors Count  Deal Terms Portfolio Exits  \
0       $7.44B             IPO               28           8               5   
1      $6.874B            None               29          12            None   
3      $3.472B        Acquired               56          13               1   
4      $4.377B        Acquired               25           5               2   
5     $571.26M            None               26           8            None   

   year joined  age  Unicorn age  
0         2017   10            5  
1         2012   20           10  
3         2011   17           11  
4         2018   31            4  
5         2018   10            4

Converting "Total Raised" Strings into integer Type

In [16]:
df['Total Raised'] 


0         $7.44B
1        $6.874B
3        $3.472B
4        $4.377B
5       $571.26M
          ...   
1032    $181.06M
1033       $700M
1034    $449.72M
1035     $525.5M
1036     $604.5M
Name: Total Raised, Length: 1015, dtype: object

In [17]:
def multiply_by_amount_raised(amt):
    try:
        amount = {'B': 1000000000, 'M': 1000000, 'K': 1000}
        return float(amt[:-1]) * amount[amt[-1:]]
    except TypeError:
        pass

In [18]:
df['Total Raised'] = df['Total Raised'].str.replace('$','')
df['Total Raised'] = df['Total Raised'].replace('None',0)
df['Total Raised'] = df['Total Raised'].apply(multiply_by_amount_raised)

/tmp/ipykernel_613377/2827083612.py:1: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  df['Total Raised'] = df['Total Raised'].str.replace('$','')


In [19]:
df.head()

Company  Valuation ($B) Date Joined        Country         City  \
0   Bytedance           140.0  2017-04-07          China      Beijing   
1      SpaceX           100.3  2012-12-01  United States    Hawthorne   
3      Klarna            45.6  2011-12-12         Sweden    Stockholm   
4  Epic Games            42.0  2018-10-26  United States         Cary   
5       Canva            40.0  2018-01-08      Australia  Surry Hills   

                       Industry  \
0       Artificial intelligence   
1                         Other   
3                       Fintech   
4                         Other   
5  Internet software & services   

                                   Select Inverstors  Founded Year  \
0  Sequoia Capital China, SIG Asia Investments, S...          2012   
1  Founders Fund, Draper Fisher Jurvetson, Rothen...          2002   
3  Institutional Venture Partners, Sequoia Capita...          2005   
4              Tencent Holdings, KKR, Smash Ventures          1991   
5  Sequoia Capital China, Blackbird Ventures, Mat...          2012   

   Total Raised Financial Stage  Investors Count  Deal Terms Portfolio Exits  \
0  7.440000e+09             IPO               28           8               5   
1  6.874000e+09            None               29          12            None   
3  3.472000e+09        Acquired               56          13               1   
4  4.377000e+09        Acquired               25           5               2   
5  5.712600e+08            None               26           8            None   

   year joined  age  Unicorn age  
0         2017   10            5  
1         2012   20           10  
3         2011   17           11  
4         2018   31            4  
5         2018   10            4

In [20]:
print(df["Industry"].value_counts())

Fintech                                203
Internet software & services           192
E-commerce & direct-to-consumer        107
Artificial intelligence                 71
Health                                  69
Supply chain, logistics, & delivery     57
Other                                   55
Cybersecurity                           49
Data management & analytics             41
Mobile & telecommunications             36
Hardware                                33
Auto & transportation                   29
Edtech                                  28
Consumer & retail                       24
Travel                                  14
Artificial Intelligence                  7
Name: Industry, dtype: int64


###### converting strinng type value to seprate column 
get_dummies() function is used to convert categorical variable into dummy/indicator variables. Data of which to get dummy indicators. String to append DataFrame column names. 

In [21]:
df= pd.get_dummies(data=df, columns=["Industry"])


### Dropping all useless column

In [22]:
useless_columns=["Company","Date Joined", "Country","City","Financial Stage","Select Inverstors","Founded Year","Portfolio Exits","year joined"]

for item in useless_columns:
    df.drop(item , axis=1, inplace=True)
    


In [23]:
df.info()
df.head()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1015 entries, 0 to 1036
Data columns (total 22 columns):
 #   Column                                        Non-Null Count  Dtype  
---  ------                                        --------------  -----  
 0   Valuation ($B)                                1015 non-null   float64
 1   Total Raised                                  991 non-null    float64
 2   Investors Count                               1015 non-null   int64  
 3   Deal Terms                                    1015 non-null   int64  
 4   age                                           1015 non-null   int64  
 5   Unicorn age                                   1015 non-null   int64  
 6   Industry_Artificial Intelligence              1015 non-null   uint8  
 7   Industry_Artificial intelligence              1015 non-null   uint8  
 8   Industry_Auto & transportation                1015 non-null   uint8  
 9   Industry_Consumer & retail                    1015 non-null   u

Valuation ($B)  Total Raised  Investors Count  Deal Terms  age  \
0           140.0  7.440000e+09               28           8   10   
1           100.3  6.874000e+09               29          12   20   
3            45.6  3.472000e+09               56          13   17   
4            42.0  4.377000e+09               25           5   31   
5            40.0  5.712600e+08               26           8   10   

   Unicorn age  Industry_Artificial Intelligence  \
0            5                                 0   
1           10                                 0   
3           11                                 0   
4            4                                 0   
5            4                                 0   

   Industry_Artificial intelligence  Industry_Auto & transportation  \
0                                 1                               0   
1                                 0                               0   
3                                 0                               0   
4                                 0                               0   
5                                 0                               0   

   Industry_Consumer & retail  ...  Industry_E-commerce & direct-to-consumer  \
0                           0  ...                                         0   
1                           0  ...                                         0   
3                           0  ...                                         0   
4                           0  ...                                         0   
5                           0  ...                                         0   

   Industry_Edtech  Industry_Fintech  Industry_Hardware  Industry_Health  \
0                0                 0                  0                0   
1                0                 0                  0                0   
3                0                 1                  0                0   
4                0                 0                  0                0   
5                0                 0                  0                0   

   Industry_Internet software & services  \
0                                      0   
1                                      0   
3                                      0   
4                                      0   
5                                      1   

   Industry_Mobile & telecommunications  Industry_Other  \
0                                     0               0   
1                                     0               1   
3                                     0               0   
4                                     0               1   
5                                     0               0   

   Industry_Supply chain, logistics, & delivery  Industry_Travel  
0                                             0                0  
1                                             0                0  
3                                             0                0  
4                                             0                0  
5                                             0                0  

[5 rows x 22 columns]

In This Stage We have all columns with numarical values, no more object type values

### Scaling all values in range of 0 to 1

In [24]:
cols_to_scale= ["Valuation ($B)","Total Raised","Investors Count","Deal Terms","age","Unicorn age"]  

from sklearn.preprocessing import MinMaxScaler
scaler = MinMaxScaler()

df[cols_to_scale]= scaler.fit_transform(df[cols_to_scale])
df.describe()

Valuation ($B)  Total Raised  Investors Count   Deal Terms  \
count     1015.000000    991.000000      1015.000000  1015.000000   
mean         0.015775      0.076594         0.157906     0.157584   
std          0.048457      0.098406         0.109246     0.114701   
min          0.000000      0.000000         0.000000     0.000000   
25%          0.000719      0.030390         0.087912     0.052632   
50%          0.004317      0.050604         0.142857     0.157895   
75%          0.015108      0.084871         0.197802     0.210526   
max          1.000000      1.000000         1.000000     1.000000   

               age  Unicorn age  Industry_Artificial Intelligence  \
count  1015.000000  1015.000000                       1015.000000   
mean      0.045312     0.145419                          0.006897   
std       0.198468     0.133541                          0.082799   
min       0.000000     0.000000                          0.000000   
25%       0.002474     0.066667                          0.000000   
50%       0.003464     0.066667                          0.000000   
75%       0.004948     0.200000                          0.000000   
max       1.000000     1.000000                          1.000000   

       Industry_Artificial intelligence  Industry_Auto & transportation  \
count                       1015.000000                     1015.000000   
mean                           0.069951                        0.028571   
std                            0.255190                        0.166681   
min                            0.000000                        0.000000   
25%                            0.000000                        0.000000   
50%                            0.000000                        0.000000   
75%                            0.000000                        0.000000   
max                            1.000000                        1.000000   

       Industry_Consumer & retail  ...  \
count                 1015.000000  ...   
mean                     0.023645  ...   
std                      0.152016  ...   
min                      0.000000  ...   
25%                      0.000000  ...   
50%                      0.000000  ...   
75%                      0.000000  ...   
max                      1.000000  ...   

       Industry_E-commerce & direct-to-consumer  Industry_Edtech  \
count                               1015.000000      1015.000000   
mean                                   0.105419         0.027586   
std                                    0.307244         0.163865   
min                                    0.000000         0.000000   
25%                                    0.000000         0.000000   
50%                                    0.000000         0.000000   
75%                                    0.000000         0.000000   
max                                    1.000000         1.000000   

       Industry_Fintech  Industry_Hardware  Industry_Health  \
count       1015.000000        1015.000000      1015.000000   
mean           0.200000           0.032512         0.067980   
std            0.400197           0.177444         0.251836   
min            0.000000           0.000000         0.000000   
25%            0.000000           0.000000         0.000000   
50%            0.000000           0.000000         0.000000   
75%            0.000000           0.000000         0.000000   
max            1.000000           1.000000         1.000000   

       Industry_Internet software & services  \
count                            1015.000000   
mean                                0.189163   
std                                 0.391831   
min                                 0.000000   
25%                                 0.000000   
50%                                 0.000000   
75%                                 0.000000   
max                                 1.000000   

       Industry_Mobile & telecommunications  Industry_Other  \
count                          

This is the final form of our data.
* 22 column , 1015 row
* all data in numarical form and between (0 to 1)

### Spliting data into Train and Test dataset

In [25]:
X = df.drop("Valuation ($B)",axis ="columns")
y = df["Valuation ($B)"]

In [26]:
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(X,y,test_size= 0.2, random_state=0)

In [27]:
for i in(X_train, X_test, y_train, y_test):
    print(i.shape)

(812, 21)
(203, 21)
(812,)
(203,)


### Building the Model
Useing keras.Sequential Model to make ANN with 21 Input layer and 1 output layer. inside it has 15 hidden layer

In [28]:
model = keras.Sequential([
    keras.layers.Dense(21, input_shape=(21,), activation='relu'),
    keras.layers.Dense(15, activation='relu'),
    keras.layers.Dense(1, activation='sigmoid')
])

# opt = keras.optimizers.Adam(learning_rate=0.01)

model.compile(optimizer='adam',
              loss='mean_squared_error',
              metrics=['accuracy'])


model.fit(X_train, y_train, epochs=10)



2022-05-06 11:22:04.741049: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcuda.so.1'; dlerror: libcuda.so.1: cannot open shared object file: No such file or directory
2022-05-06 11:22:04.741080: W tensorflow/stream_executor/cuda/cuda_driver.cc:269] failed call to cuInit: UNKNOWN ERROR (303)
2022-05-06 11:22:04.741098: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:156] kernel driver does not appear to be running on this host (csctcloud): /proc/driver/nvidia/version does not exist
2022-05-06 11:22:04.741326: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 AVX512F FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2022-05-06 11:22:04.815505: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:185] None of the M

Epoch 1/10
26/26 [==============================] - 0s 900us/step - loss: nan - accuracy: 0.2155
Epoch 2/10
26/26 [==============================] - 0s 838us/step - loss: nan - accuracy: 0.2155
Epoch 3/10
26/26 [==============================] - 0s 843us/step - loss: nan - accuracy: 0.2155
Epoch 4/10
26/26 [==============================] - 0s 848us/step - loss: nan - accuracy: 0.2155
Epoch 5/10
26/26 [==============================] - 0s 841us/step - loss: nan - accuracy: 0.2155
Epoch 6/10
26/26 [==============================] - 0s 841us/step - loss: nan - accuracy: 0.2155
Epoch 7/10
26/26 [==============================] - 0s 844us/step - loss: nan - accuracy: 0.2155
Epoch 8/10
26/26 [==============================] - 0s 835us/step - loss: nan - accuracy: 0.2155
Epoch 9/10
26/26 [==============================] - 0s 863us/step - loss: nan - accuracy: 0.2155
Epoch 10/10
26/26 [==============================] - 0s 817us/step - loss: nan - accuracy: 0.2155


In [29]:
# evaluate the keras model
_, accuracy = model.evaluate(X_test, y_test)
print('Accuracy: %.2f' % (accuracy*100))

7/7 [==============================] - 0s 863us/step - loss: nan - accuracy: 0.3005
Accuracy: 30.05


### Evaluation
With this Neural Network we get 30.05 Accuracy, which is something but not has a lot to improve

### GradientBoostingRegressor
Gradient boosting is a regression and classification machine learning technique that generates a prediction model in the form of an ensemble of weak prediction models.

In [30]:
X_train = X_train.replace((np.inf, -np.inf, np.nan), 0).reset_index(drop=True)
X_test = X_test.replace((np.inf, -np.inf, np.nan), 0).reset_index(drop=True)

In [31]:
from sklearn.ensemble import GradientBoostingRegressor

GradientBoost_model = GradientBoostingRegressor(
    n_estimators=1000, 
    learning_rate=0.1,
    random_state=0)
GradientBoost_model.fit(X_train, y_train)


GradientBoost_model.score(X_test, y_test)

0.740816285311199

Using this Model we can see it produced much better accuracy for our dataset.
### with 74% Accuracy our model is definately usable in real world Problem